### DISCUSSION TITLE
Building with .js files with no type

### DISCUSSION THREAD

**Discussion by walidvb** [vercel/community#311] [Show & Tell]
I am porting a codebase to typescript(and nextjs), and while my app builds cerrectly on my machine, I keep getting a 
`Type error: Cannot find module 'src/components/Drops/vailiens' or its corresponding type declarations.` error.
Adding `// @ts-nocheck` at the top of the .js file resolves this(but i'd have to do it for every file.)

My tsconfig is as follows. i tried various combinations of `allowJs`and `checkJs`, to no avail. Also, I barely made changes to the `create-nextjs-app` config

```json
{
  "compilerOptions": {
    "target": "es5",
    "lib": ["dom", "dom.iterable", "esnext"],
    "allowJs": true,
    "checkJs": false,
    "skipLibCheck": true,
    "strict": false,
    "forceConsistentCasingInFileNames": true,
    "noEmit": true,
    "esModuleInterop": true,
    "module": "esnext",
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "jsx": "preserve",
    "incremental": true,
    "baseUrl": ".",
    "paths": {
      "@components/*": ["src/components/*"],
    },
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"],
  "exclude": ["node_modules"]
}

```

**Comment by swarnava** [comment]
Looks like there is a typo in your tsconfig:

**Current**:

`      "@components/*": ["src/components/*"],`

**Suggested**:

`      "@/components/*": ["components/*"]`

Ref: https://nextjs.org/docs/advanced-features/module-path-aliases

**Reply by walidvb** [reply]
thanks, but this should also work with `@components`, or really, anything as a key
